### PR TITLE
Add SentiWS dataset for pos-tagging and sentiment-scoring (German)

### PR DESCRIPTION
**Pull request by harshalmittal4**

### CODE REVIEW

**Comment by harshalmittal4** [comment]
Hi @lhoestq @yjernite, requesting you to review this for any changes needed. Thanks!

**Comment by harshalmittal4** [comment]
Hi @lhoestq @yjernite, requesting you to review this for any changes needed. Thanks! :)

File: datasets/senti_ws/README.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
+---
+annotations_creators: []
+language_creators:
+- other
```
Comment (lhoestq) [inline datasets/senti_ws/README.md:4]:
```suggestion
annotations_creators:
- expert-generated
- machine-generated
language_creators:
- found
```

File: datasets/senti_ws/README.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+languages:
+- de
+licenses:
+- cc-by-sa-3.0
+multilinguality:
+- monolingual
+size_categories:
+- n<1K
+source_datasets: []
```
Comment (lhoestq) [inline datasets/senti_ws/README.md:13]:
```suggestion
source_datasets:
- original
```

File: datasets/senti_ws/README.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+For pos-tagging:
+{ 
+"word":"Abbau"
+"pos-tag": "NN"	
+}
+For sentiment-scoring:
+{
+"word":"Abbau"
+"sentiment-score":-0.058
+}
```
Comment (lhoestq) [inline datasets/senti_ws/README.md:79]:
````suggestion

For pos-tagging:

```
{ 
"word":"Abbau"
"pos_tag": 0
}
```

For sentiment-scoring:

```
{
"word":"Abbau"
"sentiment-score":-0.058
}
```
````

File: datasets/senti_ws/README.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+For pos-tagging:
+                {
+                    "word": datasets.Value("string"),
+                    "pos-tag": datasets.Value("string"),
+                }
+For sentiment-scoring:
+                {
+                    "word": datasets.Value("string"),
+                    "sentiment-score": datasets.Value("float32"),
+                }
```
Comment (lhoestq) [inline datasets/senti_ws/README.md:93]:
```suggestion
For pos-tagging:
- word: one word as a string,
- pos_tag: the part-of-speech tag of the word as an integer,

For sentiment-scoring:
- word: one word as a string,
- sentiment-score: the sentiment score of the word as a float between -1 and 1,

```

File: datasets/senti_ws/senti_ws.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+    def _info(self):
+
+        if (
+            self.config.name == "pos-tagging"
+        ):  # the pos-tags are ["NN", "VVINF", "ADJX", "ADV"] -> ["noun", "verb", "adjective", "adverb"]
+            features = datasets.Features(
+                {
+                    "word": datasets.Value("string"),
+                    "pos-tag": datasets.Value("string"),
```
Comment (lhoestq) [inline datasets/senti_ws/senti_ws.py:68]:
```suggestion
                    "pos_tag": datasets.ClassLabel(names=["NN", "VVINF", "ADJX", "ADV"]),
```
can you use ClassLabel here please and rename the column for consistency with other datasets ?

File: datasets/senti_ws/senti_ws.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        data_dir = dl_manager.download_and_extract(my_urls)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "sourcefiles": sorted(
+                        os.path.join(data_dir[0], f)
+                        for f in ["SentiWS_v2.0_Positive.txt", "SentiWS_v2.0_Negative.txt"]
+                    ),
```
Comment (lhoestq) [inline datasets/senti_ws/senti_ws.py:113]:
```suggestion
                    "sourcefiles": [
                        os.path.join(data_dir[0], f)
                        for f in ["SentiWS_v2.0_Positive.txt", "SentiWS_v2.0_Negative.txt"]
                    ],
```
you don't need to sort here

**Review by lhoestq — COMMENTED** [review]
Nice thanks !

I left a few comments.

After the changes you'll need to regenerate the dataset_infos.json file

**Comment by harshalmittal4** [comment]
Hi @lhoestq , I have updated the PR

**Review by lhoestq — APPROVED** [review]
Nice thank you !

**Pull request merged**